experimental/Schemes/IdealSheaves.jl

```diff
@@ -1210,3 +1210,105 @@ function saturation(I::IdealSheaf, J::IdealSheaf)
   return IdealSheaf(X, K, check=false)
 end
 
+
+@doc raw"""
+    IdealSheaf(X::NormalToricVariety, I::MPolyIdeal)
+
+Create a sheaf of ideals on a toric variety ``X`` from a homogeneous ideal 
+`I` in its `cox_ring`.
+
+# Examples
+```jldoctest
+julia> P3 = projective_space(NormalToricVariety, 3)
+Normal, non-affine, smooth, projective, gorenstein, fano, 3-dimensional toric variety without torusfactor
+
+julia> (x1,x2,x3,x4) = gens(cox_ring(P3))
+4-element Vector{MPolyDecRingElem{QQFieldElem, QQMPolyRingElem}}:
+ x1
+ x2
+ x3
+ x4
+
+julia> I = ideal([x2,x3])
+ideal(x2, x3)
+
+julia> IdealSheaf(P3, I)
+Sheaf of ideals
+  on scheme over QQ covered with 4 patches
+    1: [x_1_1, x_2_1, x_3_1]   normal, affine toric variety
+    2: [x_1_2, x_2_2, x_3_2]   normal, affine toric variety
+    3: [x_1_3, x_2_3, x_3_3]   normal, affine toric variety
+    4: [x_1_4, x_2_4, x_3_4]   normal, affine toric variety
+with restrictions
+  1: ideal(x_2_1, x_3_1)
+  2: ideal(x_2_2, x_3_2)
+  3: ideal(1, x_3_3)
+  4: ideal(x_2_4, 1)
+```
+"""
+function IdealSheaf(X::NormalToricVariety, I::MPolyIdeal)
+  @req base_ring(I) === cox_ring(X) "ideal must live in the cox ring of the variety"
+
+  # We currently only support this provided that the following conditions are met:
+  # 1. All maximal cones are smooth, i.e. the fan is smooth/X is smooth.
+  # 2. The dimension of all maximal cones matches the dimension of the fan.
+  @req is_smooth(X) "Currently, ideal sheaves are only supported for smooth toric varieties"
+  @req all(m -> dim(m) == dim(X), maximal_cones(X)) "Currently, ideal sheaves require that all maximal cones have the dimension of the variety"
+
+  ideal_dict = IdDict{AbsSpec, Ideal}()
+
+  # We need to dehomogenize the ideal I ⊂ S = ℂ[xᵨ| ρ ∈ Σ(1)] to the local 
+  # charts Uₛ, s ⊂ Σ a cone. To this end we use Proposition 5.2.10 from 
+  # Cox-Little-Schenck, p. 223ff. Abstractly, the chart 
+  # Uₛ is isomorphic to ℂˢ⁽¹⁾ = Spec(ℂ[xᵨ | ρ ∈ s(1)]). But these are 
+  # not the coordinates we use for the affine charts here. They 
+  # correspond to the `hilbert_basis(dual_cone(U_s))`.
+
+  # TODO: In the long run we should think about making the creation of the 
+  # following dictionary lazy. But this requires partial rewriting of the 
+  # ideal sheaves as a whole, so we postpone it for the moment.
+
+  for (k, U) in enumerate(affine_charts(X))
+
+    # We first create the morphism Φₛ* from p. 224, l. 3.
+    indices = [j for j in 1:nrays(X) if ray_indices(maximal_cones(X))[k,j]]
+    help_ring, x_rho = polynomial_ring(QQ, ["x_$j" for j in indices])
+    imgs_phi_star = [j in indices ? x_rho[findfirst(k->k==j, indices)] : one(help_ring) for j in 1:nrays(X)]
+    phi_s_star = hom(cox_ring(X), help_ring, imgs_phi_star)
+
+    # Now we need to create the inverse of α*.
+    imgs_alpha_star = elem_type(help_ring)[]
+    for m in hilbert_basis(dual_cone(U))
```

Comment:
My recollection is a bit vague - I believe this dates back probably half a year or even longer, when we first prototyped the toric schemes. The ideas did not really go very deep, but focused on organizing the code with some helper functions. Nothing more than that. This is I think why this exists in the first place.

Certainly, I am more than happy to rename it. Based on what you are saying, maybe it would be best to align it with the polymake name, i.e. `weight_cone`. Let me prepare that...

Reply:
Proposal is here 0f159c61f68590d9efcaea5bdf8769b45c4f5c59. What do you think?